backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala

```diff
@@ -98,11 +98,28 @@ object CHBackendSettings extends BackendSettingsApi with Logging {
       true
     }
 
+    // Validate if all types are supported.
+    def hasComplexType: Boolean = {
+      // Collect unsupported types.
+      val unsupportedDataTypes = fields.map(_.dataType).collect {
+        case _: MapType => "MapType"
+        case _: StructType => "StructType"
+        case _: ArrayType => "ArrayType"
+      }
+      for (unsupportedDataType <- unsupportedDataTypes) {
+        // scalastyle:off println
+        println(
```

Comment:
is the `println ` for debug purpose?

Reply:
Not really. This is necessary to print out the fallback reason directly as the info hard to pass back to FallbackReporter.

Reply:
can we use log to print the message?